### ISSUE TITLE
Error on memberdef without 'name' in 0.19.1

### ISSUE THREAD

**Issue by tim-janik** [marzer/poxy#39]
## Describe the bug

Building the [Anklang docs](https://github.com/tim-janik/anklang/) failed once we switched to Ubuntu 24.04:

https://github.com/tim-janik/anklang/actions/runs/11767824810/job/32776946290

Poxy failes to sort meberdefs in postprocessing:

```py
Post-processing XML files

*************

Traceback (most recent call last):
  File "/usr/local/lib/python3.12/dist-packages/poxy/main.py", line 623, in main
    run(
  File "/usr/local/lib/python3.12/dist-packages/poxy/run.py", line 1801, in run
    postprocess_xml(context)
  File "/usr/local/lib/python3.12/dist-packages/poxy/run.py", line 770, in postprocess_xml
    group.sort(key=sort_members_by_name)
TypeError: '<' not supported between instances of 'str' and 'NoneType'

*************

You appear to have triggered an internal bug!
Please re-run poxy with --bug-report and file an issue at github.com/marzer/poxy/issues
Many thanks!

*************
```

## Environment

The environment for this poxy run is built via the following Dockerfile:
  https://github.com/tim-janik/anklang/blob/trunk/misc/Dockerfile.noble

$ poxy --version
0.19.1

## Additional information

For now, I am applying the following patch:
```diff
--- ./usr/local/lib/python3.12/dist-packages/poxy/run.orig	2024-11-10 19:50:41.858752777 +0000
+++ ./usr/local/lib/python3.12/dist-packages/poxy/run.py	2024-11-10 19:50:50.478776979 +0000
@@ -746,7 +746,7 @@
 
                     # re-sort members to override Doxygen's weird and stupid sorting 'rules'
                     if 1:
-                        sort_members_by_name = lambda tag: tag.find(r'name').text
+                        sort_members_by_name = lambda tag: tag.find(r'name').text or ''
                         members = [tag for tag in section.findall(r'memberdef')]
                         for tag in members:
                             section.remove(tag)
```

Which works around the issue.

## `--bug-report`
-   [x ] I have attached the zip file generated when using the --bug-report option

[poxy_bug_report.zip](https://github.com/user-attachments/files/17692896/poxy_bug_report.zip)

**Comment by marzer** [comment]
Thanks for the report! Can you show me the C++ code responsible for this, or a reduced reproduction of it? I'd like to understand a bit more what is happening here before I apply your patch.

**Comment by tim-janik** [comment]
> Thanks for the report! Can you show me the C or C++ code that was parsed to cause this? Or a reduced reproduction of it? I'd like to understand a bit more what is happening here before I apply your patch.

My patch is just a workaround for our CI, I'm not claiming that it is the proper fix...
If you want to take a peak at the generated Doxygen output, you can just reporduce the steps our CI applies:
[doc/poxy.sh -b](https://github.com/tim-janik/anklang/blob/60e6ab62ba18ba8146c6217acc0c275753a5251c/.github/workflows/testing.yml#L52)

It is containerized, so you can reproduce it locally with this:

```sh
git clone --recurse-submodules  git@github.com:tim-janik/anklang.git && cd anklang
docker run -ti --rm -v $PWD:/hostwd -w /hostwd ghcr.io/tim-janik/anklang-ci:noble-latest  make -j
docker run -ti --rm -v $PWD:/hostwd -w /hostwd ghcr.io/tim-janik/anklang-ci:noble-latest  doc/poxy.sh -b
```

**Comment by marzer** [comment]
Ah, turns out there's no need. I was able to determine the cause from the logs: class `Ase::Driver` contains an enum `Flags` which (I assume) is specified using the C-style form `typedef enum { /*... */} Flags`. This causes doygen to emit the name field as simply `<name></name>`, which was not a scenario I was accounting for.

**Comment by marzer** [comment]
Should be fixed in v0.19.3, thanks :)
